src/templates/utils.js

```diff
@@ -0,0 +1,36 @@
+import tcomb from 'tcomb-validation';
+
+function toGetter(prop) {
+  const capsProp = prop.charAt(0).toUpperCase() + prop.slice(1);
+  return `get${capsProp}`;
+}
+
+/**
+ * Given a tcomb struct and a Javascript object used to instantiate that
+ * struct, create getters for all props on the tcomb struct and then
+ * instantiate it. This is intended to be used when constructing the
+ * LazyTemplateInterface, where somewhat stronger typing can propogate errors
+ * earlier.
+ *
+ * @param {LazyTemplateInterface} TemplateInterface
+ * @param {object} instance
+ * @return {LazyTemplateInterface} A lazy template interface extended with
+ * getters.
+ */
+export function constructWithGetters(TemplateInterface, instance) {
+  const interfaceProps = Object.keys(TemplateInterface.meta.props);
```

Comment:
What is `props` in this situation and how is it being used?

Reply:
They are a field on the `meta` object of an uninstantiated tcomb type: https://github.com/gcanti/tcomb/blob/master/docs/API.md#the-meta-object

That link talks about the meta object, and if you search for `props` you will see that `tcomb.interface` and `tcomb.struct` have `props` objects on their meta object.